Common/src/main/java/com/tabnineCommon/lifecycle/PluginInstalled.kt

```diff
@@ -0,0 +1,41 @@
+package com.tabnineCommon.lifecycle
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.util.messages.MessageBusConnection
+import com.intellij.util.messages.Topic
+
+fun interface PluginInstalled {
+    companion object {
+        val PLUGIN_INSTALLED_TOPIC = Topic.create(
+            "Plugin Installed Notifier",
+            PluginInstalled::class.java
+        )
+        var isNewInstallation: Boolean? = null // denotes unknown
+            set(value) {
+                // once toggled to true , don't allow  toggling back to false
```

Comment:
this is importatnt because the binary exits multiple times on install due to bootstrapping and experiment changes etc. we need to persist this fact between re-runs in the same session